crates/cli/build.rs

```diff
@@ -0,0 +1,59 @@
+use chrono::prelude::*;
+use std::fs;
+use std::process::Command;
+use std::str;
+
+fn main() {
+    // Rebuild if this build.rs file changes
+    println!("cargo:rerun-if-changed=build.rs");
+
+    // The version file is located at the root of the repository
+    let version_file_path = "../../version.txt";
+
+    // Rebuild if version file changes
+    println!("cargo:rerun-if-changed={}", version_file_path);
+
+    // Read the version file
+    let version_file_contents = fs::read_to_string(version_file_path).unwrap();
+
+    // If the version is "built-from-source", replace it with the git commit information
+    let version = match version_file_contents.trim() {
+        "built-from-source" => {
+            // Rebuild if a new Git commit is made
+            println!("cargo:rerun-if-changed=.git/HEAD");
+
+            // Get the hash of the current commit
+            let git_describe_output = Command::new("git")
+                .arg("describe")
+                .arg("--always")
+                .arg("--dirty= with additional changes") // Add a suffix if the working directory is dirty
```

Comment:
Would it pick up on these git changes given that we typically rerun if a new commit was made? So it seems like you could make a commit, rerun is triggered, the version string is set, after that you make some changes without committing and rerun is not triggered?

Reply:
I just ran `cargo run -- version` on this branch and got the normal commit hash with no extra message. Then I modified the `README.md` and ran `cargo run -- version` again, this time I got the additional changes message, so it does correctly rerun the build script.